### PR TITLE
Immutable connect

### PR DESCRIPTION
**Pull request by al8n**
use &self for connect function

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+**target/
```
Comment (danielSanchezQ) [inline .gitignore:1]: why `**`?

File: overwatch-rs/src/services/relay.rs [hunk 2]
```diff
@@ -157,7 +157,7 @@ impl<S: ServiceCore> Relay<S> {
 
     #[instrument(skip_all, err(Debug))]
     async fn handle_relay_response(
-        &mut self,
+        &self,
```
Comment (danielSanchezQ) [inline overwatch-rs/src/services/relay.rs:160]: Looks good, are there are any other `&mut` that should be just `&` ? If so please add to this pr. If not omit this :)

**Review by danielSanchezQ — APPROVED** [review]
LGTM.

There are some warnings on the `print_service` test that didn't promote to errors in the CI (@bacv ?).

If possible lets fix them as well. Otherwise can be done in a cleaning chore PR.

File: .gitignore (hunk 1 above)
Comment (al8n) [reply .gitignore:1]: Ignore all target directories in this workspace.

File: overwatch-rs/src/services/relay.rs [hunk 3]
```diff
@@ -141,13 +141,13 @@ impl<S: ServiceCore> Relay<S> {
     }
 
     #[instrument(skip(self), err(Debug))]
-    pub async fn connect(&mut self) -> Result<OutboundRelay<S::Message>, RelayError> {
```
Comment (Zeegomo) [inline overwatch-rs/src/services/relay.rs:144]: btw I think it was my bad, this could have been a `self` function from the start

Comment (danielSanchezQ) [reply overwatch-rs/src/services/relay.rs:144]: Actually yes, this should consume the relay probably.

**Pull request merged**